packages/integrations/src/adguard-home/adguard-home-integration.ts

```diff
@@ -0,0 +1,148 @@
+import { Integration } from "../base/integration";
+import { IntegrationTestConnectionError } from "../base/test-connection-error";
+import type { DnsHoleSummaryIntegration } from "../interfaces/dns-hole-summary/dns-hole-summary-integration";
+import type { DnsHoleSummary } from "../interfaces/dns-hole-summary/dns-hole-summary-types";
+import { filteringStatusSchema, statsResponseSchema, statusResponseSchema } from "./adguard-home-types";
+
+export class AdGuardHomeIntegration extends Integration implements DnsHoleSummaryIntegration {
+  public async getSummaryAsync(): Promise<DnsHoleSummary> {
+    const statsResponse = await fetch(`${this.integration.url}/control/stats`, {
+      headers: {
+        Authorization: `Basic ${this.getAuthorizationHeaderValue()}`,
+      },
+    });
+
+    if (!statsResponse.ok) {
+      throw new Error(
+        `Failed to fetch stats for ${this.integration.name} (${this.integration.id}): ${statsResponse.statusText}`,
+      );
+    }
+
+    const statusResponse = await fetch(`${this.integration.url}/control/status`, {
+      headers: {
+        Authorization: `Basic ${this.getAuthorizationHeaderValue()}`,
+      },
+    });
+
+    if (!statusResponse.ok) {
+      throw new Error(
+        `Failed to fetch status for ${this.integration.name} (${this.integration.id}): ${statusResponse.statusText}`,
+      );
+    }
+
+    const filteringStatusResponse = await fetch(`${this.integration.url}/control/filtering/status`, {
+      headers: {
+        Authorization: `Basic ${this.getAuthorizationHeaderValue()}`,
+      },
+    });
+
+    if (!filteringStatusResponse.ok) {
+      throw new Error(
+        `Failed to fetch filtering status for ${this.integration.name} (${this.integration.id}): ${filteringStatusResponse.statusText}`,
+      );
+    }
+
+    const stats = statsResponseSchema.safeParse(await statsResponse.json());
+    const status = statusResponseSchema.safeParse(await statusResponse.json());
+    const filteringStatus = filteringStatusSchema.safeParse(await filteringStatusResponse.json());
+
+    let errorMessage = `Failed to parse summary for ${this.integration.name} (${this.integration.id}):`;
+    if (!stats.success) {
+      errorMessage += `\n- Stats parsing error: ${stats.error.message}`;
+    }
+    if (!status.success) {
+      errorMessage += `\n- Status parsing error: ${status.error.message}`;
+    }
+    if (!filteringStatus.success) {
+      errorMessage += `\n- Filtering status parsing error: ${filteringStatus.error.message}`;
+    }
+    if (!stats.success || !status.success || !filteringStatus.success) {
```

Comment:
Very nice that you combined it to one error thrown. Maybe you could use an array for the error Message and check here that the length is bigger than 2. Then you ca just call `errorMessages.join("\n")`